### PR TITLE
[DPG-946] Adding ifix-migration-toolkit to egovernments IFIX code rep…

### PR DESCRIPTION
**Pull request by shashwat-egov**
…ository

### CODE REVIEW

File: utilities/ifix-migration-toolkit/.mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.4/apache-maven-3.8.4-bin.zip
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/.mvn/wrapper/maven-wrapper.properties:1]: Delete `.mvn` directory

File: utilities/ifix-migration-toolkit/LOCALSETUP.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Local Setup
+
+To setup the national-dashboard-kafka-pipeline service in your local system, clone the [Core Service repository](https://github.com/egovernments/core-services).
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/LOCALSETUP.md:3]: Remove unrelated links and names from the file

File: utilities/ifix-migration-toolkit/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+### Service Dependencies
+- NA
+
+### Swagger API Contract
+
+- NA
+
+### Functionalities
+1. Listens on persist-national-records topic and redirects records to the module specific topic names to be indexed by kafka connector.
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/README.md:16]: Either write the correct information or keep the section blank. Remove the unrelated information

File: utilities/ifix-migration-toolkit/mvnw [hunk 4]
```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/mvnw:1]: Delete this file. It is unnecessary

File: utilities/ifix-migration-toolkit/mvnw.cmd [hunk 5]
```diff
@@ -0,0 +1,188 @@
+@REM ----------------------------------------------------------------------------
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/mvnw.cmd:1]: Delete this

File: utilities/ifix-migration-toolkit/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package org.egov.nationaldashboardkafkapipeline;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+//@SpringBootTest
+class NationalDashboardKafkaPipelineApplicationTests {
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java:7]: Unrelated file. Delete it.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalEventRequest.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@NoArgsConstructor
+@Builder
+@ToString
+public class FiscalEventRequest {
+
+    @JsonProperty("requestHeader")
+    private RequestHeader requestHeader = null;
+
+    @JsonProperty("fiscalEvent")
+    private FiscalEvent fiscalEvent = null;
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalEventRequest.java:19]: Isn't this a list?

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/PersisterDepartmentEntity.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    private String code = null;
+
+    @JsonProperty("name")
+    private String name = null;
+
+    @JsonProperty("hierarchyLevel")
+    private Integer hierarchyLevel = null;
+
+    @JsonProperty("children")
+    private String children = null;
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/PersisterDepartmentEntity.java:33]: Need to update as per the changes in ifix-department-entity-service.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/ProducerPOJO.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@ToString
+public class ProducerPOJO {
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/ProducerPOJO.java:16]: Rename as per the comment in fiscal-event-service PR.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/service/MigrationService.java [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+
+@Slf4j
+@Service
+public class MigrationService {
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/service/MigrationService.java:22]: Rename this to "FiscalEventMigrationService".

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/util/MigrationEnum.java [hunk 11]
```diff
@@ -0,0 +1,6 @@
+package org.egov.ifixmigrationtoolkit.util;
+
+public enum MigrationEnum {
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/util/MigrationEnum.java:3]: If this is specific only to Department Entity Migration, then use the prefix Departmententity and name it "DepartmentEntityMigrationEnum".

File: utilities/ifix-migration-toolkit/src/main/resources/db/migration/main/V20220913185050__ifix_fiscal_events_migration.sql [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  id CHARACTER VARYING (128) NOT NULL,
+  tenantid CHARACTER VARYING (128) NOT NULL,
+  pagenumber bigint NOT NULL,
+  batchsize bigint NOT NULL,
+  totalnumberofrecordsmigrated bigint NOT NULL,
+  service_type character varying(255),
+  createdtime bigint NOT NULL,
+
+  CONSTRAINT pk_ifix_dept_entity_migration_id PRIMARY KEY(id)
+);
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/resources/db/migration/main/V20220913185050__ifix_fiscal_events_migration.sql:24]: Add indexes

File: utilities/ifix-migration-toolkit/.mvn/wrapper/maven-wrapper.properties (hunk 1 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/.mvn/wrapper/maven-wrapper.properties:1]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/LOCALSETUP.md (hunk 2 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/LOCALSETUP.md:3]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/README.md (hunk 3 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/README.md:16]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/mvnw (hunk 4 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/mvnw:1]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/mvnw.cmd (hunk 5 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/mvnw.cmd:1]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/pom.xml [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.2.13.RELEASE</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>org.egov</groupId>
+	<artifactId>ifix-migration-toolkit</artifactId>
+	<version>1.0.0-SNAPSHOT</version>
+	<name>ifix-es-pipeline</name>
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/pom.xml:14]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java (hunk 6 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java:7]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/config/IESPConfiguration.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+@Data
+@Import({TracerConfiguration.class})
+@NoArgsConstructor
+@AllArgsConstructor
+public class IESPConfiguration {
+    @Value("${app.timezone}")
+    private String timeZone;
+
+    @PostConstruct
+    public void initialize() {
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/config/IESPConfiguration.java:28]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalClientInput.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@ToString
+public class FiscalClientInput {
+
+    @JsonProperty("topic")
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalClientInput.java:14]: This is just a model class for testing, no changes are required.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalEventRequest.java (hunk 7 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/models/FiscalEventRequest.java:19]: mongodb fiscal event service has request with a single fiscalEvent field instead of a list.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/service/MigrationService.java (hunk 10 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/service/MigrationService.java:24]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/util/MigrationEnum.java (hunk 11 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/util/MigrationEnum.java:3]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/main/resources/db/migration/main/V20220913185050__ifix_fiscal_events_migration.sql (hunk 12 above)
Comment (shashwat-egov) [reply utilities/ifix-migration-toolkit/src/main/resources/db/migration/main/V20220913185050__ifix_fiscal_events_migration.sql:24]: minor change, to be taken up as part of code cleanup.

File: utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/controller/MigrationController.java [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+    private DepartmentHierarchyLevelMigrationService departmentHierarchyLevelMigrationService;
+
+    @Autowired
+    private MasterDataMigrationService masterDataMigrationService;
+
+    @RequestMapping(value="/fiscal_event/v1/_migrate", method = RequestMethod.POST)
+    public ResponseEntity<?> migrateDataToES(@RequestBody @Valid MigrationRequest request) throws JsonProcessingException {
+        migrationService.migrateData(request);
+        Map<String, Object> responseMap = new HashMap<>();
+        responseMap.put("SUCCESS", "Department Entity data migration job created successfully");
```
Comment (rushang7-eGov) [inline utilities/ifix-migration-toolkit/src/main/java/org/egov/ifixmigrationtoolkit/controller/MigrationController.java:47]: Correct the name. It is fiscal event and not department entity

**Review by rushang7-eGov — APPROVED** [review]
